packages/@aws-cdk/integ-tests/lib/assertions/providers/lambda-handler/assertion.ts

```diff
@@ -1,34 +1,151 @@
 /* eslint-disable no-console */
-import * as assert from 'assert';
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { Match, Matcher } from '@aws-cdk/assertions/lib/helpers-internal';
 import { CustomResourceHandler } from './base';
-import { AssertionRequest, AssertionResult } from './types';
+import { AssertionResult, AssertionRequest } from './types';
 
 export class AssertionHandler extends CustomResourceHandler<AssertionRequest, AssertionResult> {
   protected async processEvent(request: AssertionRequest): Promise<AssertionResult | undefined> {
+    let actual = decodeCall(request.actual);
+    const expected = decodeCall(request.expected);
     let result: AssertionResult;
-    switch (request.assertionType) {
-      case 'equals':
-        console.log(`Testing equality between ${JSON.stringify(request.actual)} and ${JSON.stringify(request.expected)}`);
-        try {
-          assert.deepStrictEqual(request.actual, request.expected);
-          result = { data: { status: 'pass' } };
-        } catch (e) {
-          if (e instanceof assert.AssertionError) {
-            result = {
-              data: {
-                status: 'fail',
-                message: e.message,
-              },
-            };
-          } else {
-            throw e;
-          }
-        }
+    const matcher = new MatchCreator(expected).getMatcher();
+    console.log(`Testing equality between ${JSON.stringify(request.actual)} and ${JSON.stringify(request.expected)}`);
+
+    const matchResult = matcher.test(actual);
+    matchResult.finished();
+    if (matchResult.hasFailed()) {
+      result = {
+        data: JSON.stringify({
+          status: 'fail',
+          message: [
+            ...matchResult.toHumanStrings(),
+            JSON.stringify(matchResult.target, undefined, 2),
+          ].join('\n'),
+        }),
+      };
+      if (request.reportFailure) {
+        throw new Error(result.data);
```

Comment:
Is this `throw` going to be shown with the error message in the stack deployment? Do we run this code through the custom resource provider?